evm_arithmetization/src/fixed_recursive_verifier.rs

```diff
@@ -1657,39 +1657,25 @@ where
     ) -> anyhow::Result<(ProofWithPublicInputs<F, C, D>, PublicValues)> {
         let mut agg_inputs = PartialWitness::new();
 
-        agg_inputs.set_bool_target(self.segment_aggregation.lhs.is_agg, lhs_is_agg);
-        // If the lhs is not an aggregation, we set the cyclic vk to a dummy value, so
-        // that it corresponds to the aggregation cyclic vk.
-        if lhs_is_agg {
-            agg_inputs
-                .set_proof_with_pis_target(&self.segment_aggregation.lhs.agg_proof, lhs_proof);
-        } else {
-            Self::set_dummy_proof_with_cyclic_vk_pis(
-                &self.segment_aggregation.circuit,
-                &mut agg_inputs,
-                &self.segment_aggregation.lhs.agg_proof,
-                lhs_proof,
-            )
-        }
-        agg_inputs
-            .set_proof_with_pis_target(&self.segment_aggregation.lhs.segment_proof, lhs_proof);
-
-        agg_inputs.set_bool_target(self.segment_aggregation.rhs.is_agg, rhs_is_agg);
-        // If the rhs is not an aggregation, we set the cyclic vk to a dummy value, so
-        // that it corresponds to the aggregation cyclic vk.
-        if rhs_is_agg {
-            agg_inputs
-                .set_proof_with_pis_target(&self.segment_aggregation.rhs.agg_proof, rhs_proof);
-        } else {
-            Self::set_dummy_proof_with_cyclic_vk_pis(
-                &self.segment_aggregation.circuit,
-                &mut agg_inputs,
-                &self.segment_aggregation.rhs.agg_proof,
-                rhs_proof,
-            )
-        }
-        agg_inputs
-            .set_proof_with_pis_target(&self.segment_aggregation.rhs.segment_proof, rhs_proof);
+        Self::set_dummy_if_necessary(
+            self.segment_aggregation.lhs.is_agg,
+            lhs_is_agg,
+            &self.segment_aggregation.circuit,
+            &mut agg_inputs,
+            &self.segment_aggregation.lhs.segment_proof,
+            &self.segment_aggregation.lhs.agg_proof,
+            lhs_proof,
```

Comment:
The method could take `AggregationChildTarget` as argument to group `is_agg`, `segment_proof`, `agg_proof` into a single method argument.

Reply:
But that would mean having two methods (or a new `AggregationChildTarget` enum that is either a `SegmentAggregationChild` or a `TxnChildAggregationChild`)

Reply:
You don't need an enum, they are semantically the same. Let me open a PR.

Reply:
https://github.com/0xPolygonZero/zk_evm/pull/188

Reply:
Ah sorry, I had misunderstood what you meant. Indeed, they are the same, we had actually introduced the difference for clarity (to differentiate between segment aggregations and transaction aggregations even if they are both aggregations).